packages/core/src/shipping/shipping-country-action-creator.ts

```diff
@@ -1,20 +1,27 @@
 import { createAction, createErrorAction } from '@bigcommerce/data-store';
 import { Observable, Observer } from 'rxjs';
 
+import CheckoutStore from '../checkout/checkout-store';
 import { RequestOptions } from '../common/http-request';
 
 import { LoadShippingCountriesAction, ShippingCountryActionType } from './shipping-country-actions';
 import ShippingCountryRequestSender from './shipping-country-request-sender';
 
 export default class ShippingCountryActionCreator {
-    constructor(private _shippingCountryRequestSender: ShippingCountryRequestSender) {}
+    constructor(
+        private _shippingCountryRequestSender: ShippingCountryRequestSender,
+        private _store: CheckoutStore,
+    ) {}
 
     loadCountries(options?: RequestOptions): Observable<LoadShippingCountriesAction> {
+        const { checkout } = this._store.getState();
+        const { channelId } = checkout.getCheckoutOrThrow();
```

Comment:
Are there any cases, when we can load shipping countries without having a checkout? @animesh1987 

My question is all about, does this lie by throwing an error brakes any flow?

Reply:
This should cover all the clients, @agchan12 does it make sense to wrap this in an experiment?

Reply:
@serhii-tkachenko we don't load countries without checkout id, as the only anomaly I could think was payment buttons but we do load checkout in the initialise methods for those.

Reply:
This will be wrapped in an experiment on the BCApp side of things. The idea is that checkout-sdk-js will send the `?channel_id={channelId}` query parameter through to the `shipping/countries` endpoint where the experiment will live.

We will only consume the `?channel_id={channel_id}` query parameter when the experiment is on.